TestJobManagerCreateJob met nil pointer error

### What did you do?

run unit test of engine module

ref: https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftiflow%2Fghpr_verify/detail/ghpr_verify/1118/pipeline/164

detail logs: 
[do.pingcap.net.txt](https://github.com/pingcap/tiflow/files/10359590/do.pingcap.net.txt)


### What did you expect to see?

_No response_

### What did you see instead?

```go

=== Failed
=== FAIL: engine/servermaster TestJobManagerCreateJob (0.47s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x663d0f8]
goroutine 281 [running]:
testing.tRunner.func1.2({0x6b13c40, 0xafe6c40})
	/usr/local/go/src/testing/testing.go:1396 +0x372
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1399 +0x5f0
panic({0x6b13c40, 0xafe6c40})
	/usr/local/go/src/runtime/panic.go:890 +0x262
github.com/pingcap/tiflow/engine/servermaster.(*JobManagerImpl).OnWorkerDispatched(0xc00117a000, {0x8c0bd78, 0xc000b32c60}, {0x8bddf00, 0xc001d90150})
	/home/jenkins/agent/workspace/pingcap/tiflow/ghpr_verify/tiflow/engine/servermaster/jobmanager.go:716 +0x478
github.com/pingcap/tiflow/engine/framework.(*DefaultBaseMaster).doInit.func4({0xc001efda90?, 0xc001efd988?}, {0x8c0bd78, 0xc000b32c60}, {0x8bddf00, 0xc001d90150})
	/home/jenkins/agent/workspace/pingcap/tiflow/ghpr_verify/tiflow/engine/framework/master.go:431 +0x74
github.com/pingcap/tiflow/engine/framework/internal/master.(*WorkerManager).Tick(0xc0019be2a0, {0x8c0be20, 0xc00239e600})
	/home/jenkins/agent/workspace/pingcap/tiflow/ghpr_verify/tiflow/engine/framework/internal/master/worker_manager.go:338 +0x52e
github.com/pingcap/tiflow/engine/framework.(*DefaultBaseMaster).doPoll(0xc001bbe540, {0x8c0be20, 0xc00239e600})
	/home/jenkins/agent/workspace/pingcap/tiflow/ghpr_verify/tiflow/engine/framework/master.go:550 +0x147
github.com/pingcap/tiflow/engine/framework.(*DefaultBaseMaster).Poll(0xc001bbe540, {0x8c0ac68, 0xc00111e200})
	/home/jenkins/agent/workspace/pingcap/tiflow/ghpr_verify/tiflow/engine/framework/master.go:524 +0xe5
github.com/pingcap/tiflow/engine/servermaster.TestJobManagerCreateJob(0x0?)
	/home/jenkins/agent/workspace/pingcap/tiflow/ghpr_verify/tiflow/engine/servermaster/jobmanager_test.go:94 +0x833
testing.tRunner(0xc0017016c0, 0x723ff80)
	/usr/local/go/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1493 +0x75e
```

### Versions of the cluster

Dataflow Engine version (run `tiflow version`):

master@https://github.com/pingcap/tiflow/commit/857e7cfbe74620deb1a46a1f4cdfe383d4977fd1

--- a/engine/servermaster/jobmanager_test.go
+++ b/engine/servermaster/jobmanager_test.go
@@ -91,8 +91,6 @@ func TestJobManagerCreateJob(t *testing.T) {
 	}
 	job, err := mgr.CreateJob(ctx, req)
 	require.NoError(t, err)
-	err = mockMaster.Poll(ctx)
-	require.NoError(t, err)
 
 	require.Eventually(t, func() bool {
 		return mgr.JobFsm.QueryJob(job.Id) != nil

